### ISSUE TITLE
Language Server Protocol ServerCapabilities.codeLensProvider is illegal type

### ISSUE THREAD

**Issue by Taywee** [godotengine/godot#40275]
**Godot version:**

3.2.2

**OS/device including version:**

Linux

**Issue description:**

When running neovim with the LanguageClient-neovim, I get an instant error: "failed to parse semantic scopes: invalid type: boolean `false`, expected struct CodeLensOptions".  Examining the code, it is expecting a nullable codeLensProvider, which takes me to [modules/gdscript/language_server/lsp.hpp line 1664](https://github.com/godotengine/godot/blob/d0cea291d2b179610f20fb7136b3e74f26621e92/modules/gdscript/language_server/lsp.hpp#L1664).  The [specification for LSP shows that this type is optional, but may not be boolean](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#initialize):

```yaml
codeLensProvider?: CodeLensOptions;
```

Note that the other `false` fields in there are correct, as those are optionally boolean:

```yaml
colorProvider?: boolean | DocumentColorOptions | DocumentColorRegistrationOptions;
foldingRangeProvider?: boolean | FoldingRangeOptions | FoldingRangeRegistrationOptions;
```

I'm not sure why this inconsistency exists, but if you simply remove the codeLinkProvider, it should fix this.  This probably works in most dynamic language implementations, which would usually be satisfied with the falsiness.  LanguageClient-neovim is Rust-based, so the attempt to deserialize it fails.

I'll see if I can get the program building and try to open a pull request for this if I can fix the issue just by commenting out that line.

**Steps to reproduce:**

Try to connect with LanguageClient-neovim.

**Minimal reproduction project:**

Any project with gdscript should trigger it.

**Comment by Taywee** [comment]
I was able to fix that particular issue just by commenting out the line, but I couldn't get it fully working due to a related type issue in a LanguageClient-neovim dependent library.  I'm going to give that its own bug report now.

**Comment by Taywee** [comment]
Opened #40276 for this.

**Comment by kovasap** [comment]
Thanks for finding and fixing this, just saved me a bunch of time!

**Comment by akien-mga** [comment]
Fixed by #40276.
